### PR TITLE
SharedOT: Reusable base class for hosting OT types

### PR DESCRIPTION
**Pull request by DLehenbauer**
This is a refactoring of the /experimental/ SharedOT type to create a reusable base class for hosting OT types.

Includes two examples:
1.  Hosting 'ottypes/json1' (from ShareJS)
2. Hosting 'quill-delta' (as test case)

### CODE REVIEW

File: experimental/dds/ot/ot/src/test/delta.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+/*!
```
Comment (DLehenbauer) [inline experimental/dds/ot/ot/src/test/delta.ts:1]: @SamBroner  - FYI - SharedOT hosting 'quill-delta'

File: experimental/dds/ot/ot/src/test/ot.stress.spec.ts [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+import {
+    MockFluidDataStoreRuntime,
+    MockStorage,
+    MockContainerRuntimeFactoryForReconnection,
+    MockContainerRuntimeForReconnection,
+} from "@fluidframework/test-runtime-utils";
+import { SharedDelta } from "./delta";
+
+describe("SharedOT", () => {
+    describe("stress", () => {
```
Comment (DLehenbauer) [inline experimental/dds/ot/ot/src/test/ot.stress.spec.ts:18]: @SamBroner - Stress tests SharedOT hosting of 'quill-delta'

File: experimental/dds/ot/ot/src/ot.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    /**
+     * Queue of sequenced ops that are above minSeq.  Used by 'processCore' to
+     * adjust incoming ops to account for prior ops that the sender didn't know about
+     * at the time they submitted thier op.
+     */
+    private readonly sequencedOps: ISequencedOpInfo<TOp>[] = [];
+
+    /**
+     * Queue of local pending ops that have not yet been ACKed by the service.  Used
+     * to lazily rebuild the 'local' state cache when it is invalidated by interleaved
```
[the remark this answers is not in the document]
Comment (DLehenbauer) [reply experimental/dds/ot/ot/src/ot.ts]: @tylerbutler - we sometimes use local/global informally to refer to pending local changes vs. changes that have been sequenced by the service and therefore are global/shared knowledge.

It's an imprecise analogy to local (private) vs. global (shared) state, but is convenient short-hand.

**Review by ksbrar — APPROVED** [review]

**Comment by DLehenbauer** [comment]
Thanks, @ksbrar!

**Pull request merged**